Join table

Fadahunsiseyi opened the pull request:
- Understand the different types of relationships between tables.
- Prepare complex queries that answer analytical questions.
- Use primary key & foreign key mechanisms


![Screenshot (333)](https://user-images.githubusercontent.com/63434427/190859723-f6818bb2-234c-4024-88c3-38c2373a40af.png)
![Screenshot (334)](https://user-images.githubusercontent.com/63434427/190859725-6fae7867-0423-4d4b-9893-3c5802c9b8e8.png)
![Screenshot (335)](https://user-images.githubusercontent.com/63434427/190859726-95756ce3-8387-484f-9f6b-711ea7382c2f.png)
![Screenshot (336)](https://user-images.githubusercontent.com/63434427/190859728-13f04a1d-ce76-4f13-95c6-a08b2ec25fcb.png)
![Screenshot (337)](https://user-images.githubusercontent.com/63434427/190859729-b2f80a5b-2b42-48b6-a050-52d2291d1051.png)
![Screenshot (338)](https://user-images.githubusercontent.com/63434427/190859730-3ba0a650-284a-4821-a7eb-e5c1309595dd.png)
 for joining tables.

Fadahunsiseyi committed:
Create the vets table

Fadahunsiseyi committed:
add primary key to the vets id

Fadahunsiseyi committed:
Create the specializations table

Fadahunsiseyi committed:
Create the visits table

Fadahunsiseyi committed:
Add values into the vets table

Fadahunsiseyi committed:
Add values into the specializarions  table

Fadahunsiseyi committed:
Add values into the visits table

Fadahunsiseyi committed:
Add queries

AnselemOdims commented on schema.sql:
```diff
@@ -20,6 +20,28 @@ ALTER TABLE species ADD PRIMARY KEY(id);
 
 
-CREATE TABLE species(id INT GENERATED ALWAYS AS IDENTITY, name CHAR);
+vet_clinic=# CREATE TABLE vets (
+vet_clinic(# id INT GENERATED ALWAYS AS IDENTITY,
+vet_clinic(# name VARCHAR,
+vet_clinic(# age INT,
+vet_clinic(# date_of_graduation DATE);
+CREATE TABLE
+
+vet_clinic=# ALTER TABLE vets ADD PRIMARY KEY(id);
+ALTER TABLE
+
+vet_clinic=# create table specializations (
+vet_clinic(#     vet_id int references vets(id),
+vet_clinic(#     species_id int references species(id)
+vet_clinic(# );
+CREATE TABLE
+
+vet_clinic=# create table visits (
+vet_clinic(# vet_id int references vets(id),
+vet_clinic(# animals_id int references animals(id),
+vet_clinic(# date_of_visit date
+vet_clinic(# );
```
- [ ] Please kindly include only the queries instead of the terminal prompt

AnselemOdims commented on queries.sql:
```diff
@@ -169,4 +169,122 @@ vet_clinic-# animals.owner_id = owners.id group by owners.full_name order by cou
+(5 rows)
+
+
+vet_clinic=# select visits.date_of_visit, animals.name, vets.name from animals
+vet_clinic-# join visits on animals.id = visits.animals_id
+vet_clinic-# join vets on visits.vet_id = vets.id
+vet_clinic-# where vets.name = 'William Tatcher'
+vet_clinic-# order by visits.date_of_visit desc limit 1;
+ date_of_visit |  name   |      name
+---------------+---------+-----------------
+ 2021-01-11    | Blossom | William Tatcher
+(1 row)
+
+
+vet_clinic=# select visits.date_of_visit, animals.name, vets.name from animals
+vet_clinic-# join visits on animals.id = visits.animals_id
+vet_clinic-# join vets on visits.vet_id = vets.id
+vet_clinic-# where vets.name = 'William Tatcher'
+vet_clinic-# order by visits.date_of_visit desc limit 1;
+ date_of_visit |  name   |      name
+---------------+---------+-----------------
+ 2021-01-11    | Blossom | William Tatcher
+(1 row)
+
+
+vet_clinic=# select vets.name, species.name from vets
+vet_clinic-# left join specializations on specializations.vet_id = vets.id
+vet_clinic-# left join species on specializations.species_id = species.id;
+       name       |                         name
+------------------+------------------------------------------------------
+ William Tatcher  | Pokemon
```
- [ ] Good job working on this, but please as stated above kindly include only the queries and not the terminal prompt and output

Fadahunsiseyi replied:
@AnselemOdims I have removed all of the command prompt outputs.

AnselemOdims reviewed (changes requested):
### Hi @Fadahunsiseyi , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

Fadahunsiseyi committed:
add only queries

muneebulrehman reviewed (approved):
Hi @Fadahunsiseyi , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Good follow-up with the previous reviewer.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Fadahunsiseyi commented:
@muneebulrehman Thank you, but my dashboard is still saying changes required.

muneebulrehman commented:
> @muneebulrehman Thank you, but my dashboard is still saying changes required.

That is because I requested changes in accordance with the previous reviewer. Maybe i mistakenly clicked on Approve in github but as you can see in the summary, it says "Required changes"

Fadahunsiseyi commented:
@muneebulrehman I have removed all of the terminal prompts. Can you please highlight what you're referring to?

Fadahunsiseyi commented:
> I have removed all of the terminal prompts. Can you please highlight what you're referring to?

 I have removed all of the terminal prompts. Can you please highlight what you're referring to?

Fadahunsiseyi commented:
> > @muneebulrehman Thank you, but my dashboard is still saying changes required.
> 
> That is because I requested changes in accordance with the previous reviewer. Maybe i mistakenly clicked on Approve in github but as you can see in the summary, it says "Required changes"

 I have removed all of the terminal prompts. Can you please highlight what you're referring to?

Fadahunsiseyi committed:
Remove terminal tables

Fadahunsiseyi committed:
Remove tables in the terminal

elmar8287 reviewed (approved):
Hi, @Fadahunsiseyi.

# APPROVED  👍 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Congratulations! 🎉 

### Highlights

- Done with queries 👍 

### Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

![](https://user-images.githubusercontent.com/64173189/128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c.gif)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
### Please, remember to tag me (@elmar8287) in your question so I can receive the notification 👍 
![giphy](https://user-images.githubusercontent.com/49064106/180751457-52107e57-31fd-486c-97fe-bcccf4e449af.gif)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Fadahunsiseyi commented:
@elmar8287 Thank you.
